manifests/init.pp

```diff
@@ -100,11 +102,17 @@
   Optional[String] $systemd_beat_log_opts_override                    = undef,
   String $systemd_beat_log_opts_template                              = $filebeat::params::systemd_beat_log_opts_template,
   String $systemd_override_dir                                        = $filebeat::params::systemd_override_dir,
-
+  Variant[String, Enum['exec', 'chocolatey']] $package_provider       = undef,
```

Comment:
Interesting, I would expect this to use exec by default as one of the params. `$filebeat::params::package_provider`, but I can see why maybe not drop it into the params since it would not be platform agnostic

Reply:
Yes I want exec by default and chocolatey as an option. Not sure what I can do here ? I want to let the choice to the user. what do you recommend ?